flink-metrics/flink-metrics-slf4j/src/test/java/org/apache/flink/metrics/slf4j/Slf4jReporterTest.java

```diff
@@ -57,10 +58,75 @@ static void setUp() {
                 TestMetricGroup.newBuilder()
                         .setMetricIdentifierFunction((s, characterFilter) -> SCOPE + delimiter + s)
                         .build();
+    }
+
+    @BeforeEach
+    void setUpReporter() {
         reporter = new Slf4jReporter();
         reporter.open(new MetricConfig());
     }
 
+    @Test
+    void testSkipOnNoMetrics() {
+        reporter.report();
+
+        assertThat(testLoggerResource.getMessages())
+                .noneMatch(logOutput -> logOutput.contains("Starting metrics report"))
+                .anyMatch(logOutput -> logOutput.contains("Skipping metrics report"));
+    }
+
+    @Test
+    void testSkipCountersIfNoCounterRegistered() {
```

Comment:
```suggestion
    void testOnlyCounterRegistered() {
```
Isn't the test method name misleading? It's doing the opposite of what the name implies. 
This applies to the other test methods below.

Reply:
whoops :sweat_smile: